### ISSUE TITLE
aws_security_group:  using same port/protocol in multiple inline rules causes state problems

### ISSUE THREAD

**Issue by joshuabaird** [terraform-providers/terraform-provider-aws#2379]
### Terraform Version
0.10.3

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_security_group

### Terraform Configuration Files
```hcl
resource "aws_security_group" "rds-private-replica" {
  name        = "blah"
  description = "blah"
  vpc_id      = "${aws_vpc.primary.id}"
  
  ingress {
    from_port       = 5432
    to_port         = 5432
    protocol        = "tcp"
    security_groups = ["${aws_security_group.ecs_sg.id}", "${aws_security_group.console_cluster.id}"]
    description     = "ECS Clusters"
  }

  ingress { 
    from_port       = 5432
    to_port         = 5432
    protocol        = "tcp"
    security_groups = ["${module.ecs-openvpn.ecs_sg}"]
    description     = "Eng VPN"
  }
```

### Expected Behavior
Two rules should be created.

### Actual Behavior
The rules are created, but Terraform detects a diff during every plan/apply and tries to apply changes.

### References
It sounds like this was a known bug and was once fixed, but we are seeing this problem in 0.10.3.  See the following issues:

* https://github.com/hashicorp/terraform/issues/2471
* https://github.com/hashicorp/terraform/issues/3066
* https://github.com/hashicorp/terraform/pull/5100

**Comment by Seldaek** [comment]
Note that  using https://www.terraform.io/docs/providers/aws/r/security_group_rule.html to define the rules outside of the aws_security_group seems to be a valid workaround. I just had to manually delete all ingress/egress rules in the AWS console to get it to apply correctly but then it stops trying to change it every single time.

**Comment by yardensachs** [comment]
Any update on this issue?

**Comment by bflad** [comment]
Hi folks 👋  Sorry this has been a longstanding issue with the AWS provider. The fix for this should be contained in #4416 which will be released with v1.19.0 of the AWS provider, likely middle of next week.

Shout outs to @loivis (and @svanharmelen who submitted an earlier, likely correct PR, which I admittedly should have reviewed and merged sooner: #3628)

Given there were so many various issues surrounding this bug, I will be locking this issue (amongst all the others) to encourage any lingering issues/discussion to be fully described in new issue(s) for consolidation. Thanks for your understanding.
